ansible/playbooks/adhoc/grow_root_vg/grow_root_vg.yml

```diff
@@ -0,0 +1,156 @@
+
+  - debug:
+      var: rootvg_pv
+
+  - name: fail if we don't find a single rootvg physical volume
+    fail:
+      msg:  Unable to find single rootvg physical volume. Please investigate manually.
+    when: rootvg_pv.stdout_lines|length != 1
+
+  - name: get list of volumes from AWS
+    delegate_to: localhost
+    ec2_vol:
+      state: list
+      instance: "{{ ec2_id }}"
+      region: "{{ ec2_region }}"
+    register: attached_volumes
+
+  - debug: var=attached_volumes
+
+  - name: set rootvg volume attributes
+    set_fact:
+      rootvg: "{{ rootvg_pv.stdout | vol_attrs }}"
+ 
+  - debug: var=rootvg
+
+  - name: get volume id of current rootvg volume
+    set_fact:
+      rootvg_volume_id: "{{ attached_volumes.volumes | get_volume_id_by_linux_device(rootvg.device) }}"
+
+  - debug: var=rootvg_volume_id
+
+  - fail:
+      msg: "Unable to look up the volume ID of {{ rootvg.device }}"
+    when: rootvg_volume_id is undefined
+
+  - name: get volume size of current rootvg volume
+    set_fact:
+      rootvg_volume_size: "{{ attached_volumes.volumes | get_volume_size_by_linux_device(rootvg.device) }}"
+
+  - debug: var=rootvg_volume_size
+
+  - name: find rootvg disk size before change 
+    shell: "awk '/{{ rootvg.bare_device }}/ {if ($2==0) print $3}' /proc/partitions"
+    register: orig_pv_size
+
+  - debug: var=orig_pv_size
+
+  - name: Take snapshot
+    delegate_to: localhost
+    ec2_snapshot:
+      region: "{{ ec2_region }}"
+      volume_id: "{{ rootvg_volume_id }}"
+      description: "Snapshot of {{ cli_tag_name }} {{ rootvg.device }} before resize"
+      wait: no ## don't wait for the snapshot to complete pushing to S3, it still exists and waiting can take hours
+    when: cli_volume_size < rootvg_volume_size
+
+  - name: Resize EBS volume 
+    delegate_to: localhost
+    ## afaict the ansible ec2 modules don't offer resize
+    command: "aws ec2 modify-volume --volume-id {{ rootvg_volume_id }} --size {{ cli_volume_size }} --region {{ ec2_region }}"
+    when: cli_volume_size < rootvg_volume_size
+    register: resize_volume
+
+  - debug: var=resize_volume
+
+  - wait_for: 
+      timeout: 10 
+    when: cli_volume_size < rootvg_volume_size
+
+  - name: Fail when problems extending
+    fail:
+      msg: "Failed to resize volume msg: {{ resize_volume.msg }}"
+    when: resize_volume.msg is defined
+
+  - name: Run partprobe to see if the size changed 
+    command: partprobe "{{ rootvg.device }}"
+
+  - name: find rootvg disk size after change
+    shell: "awk '/{{ rootvg.bare_device }}/ {if ($2==0) print $3}' /proc/partitions"
+    register: new_pv_size
+
+  - debug: var=new_pv_size
+
+  - name: check if rootvg disk size changed
+    debug:
+      msg: Rootvg disk size was not changed
+    when: orig_pv_size.stdout == new_pv_size.stdout
+
+  - name: Run growpart on last partition 
+    shell: "growpart {{ rootvg.device }} $(grep -c {{ rootvg.bare_device }}[0-9] /proc/partitions) -v"
+    register: growpart_output
```

Comment:
The subshell here makes me nervous. I'd move that out into its own task. But that's just my personal preference.